ledger/src/ui/test/bootloader/mock/mock.h

```diff
@@ -0,0 +1,91 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2021 RSK Labs Ltd
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+#ifndef _MOCK_H
+#define _MOCK_H
+
+#include <assert.h>
+#include <stdint.h>
+#include "os_exceptions.h"
+
+#define CHANNEL_APDU 0
+#define INS_ATTESTATION 0x50
+#define INS_SIGNER_AUTHORIZATION 0x51
+
+#define IO_APDU_BUFFER_SIZE (5 + 255)
+extern unsigned char G_io_apdu_buffer[IO_APDU_BUFFER_SIZE];
+
+// Assert helpers
+#define ASSERT_EQUALS(a, b) assert((a) == (b))
+#define ASSERT_TRUE(cond) assert(cond)
+#define ASSERT_FALSE(cond) assert(!(cond))
+#define ASSERT_NOT_NULL(obj) assert(NULL != obj)
+#define ASSERT_APDU_AT(i, val) assert(APDU_AT(i) == val)
+#define ASSERT_FAIL() assert(false)
```

Comment:
This is good, but I think we could benefit from richer helpers. So for example, instead of

```
ASSERT_APDU_AT(0, 0xAA);
ASSERT_APDU_AT(1, 0xBB);
ASSERT_APDU_AT(2, 0xCC);
```

we could have:

```
assert_apdu("aabbcc");
```

Also we don't necessarily need macros here, we are free to do whatever we want. I also don't think `ASSERT_EQUALS` and all of the very simple helpers really provide much of an advantage (e.g., `assert(!!cond)` or `assert(!cond)` are readable enough by themselves).

Lastly I think we could also add helpers for tasks like setting the APDU from a string, and we could also place all these helpers in a common directory so that we can share them across tests.